src/main/java/bio/terra/common/retry/SamRetry.java

```diff
@@ -0,0 +1,194 @@
+package bio.terra.common.retry;
+
+import static java.time.Instant.now;
+
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.common.exception.ConflictException;
+import bio.terra.common.exception.ErrorReportException;
+import bio.terra.common.exception.ForbiddenException;
+import bio.terra.common.exception.InternalServerErrorException;
+import bio.terra.common.exception.NotFoundException;
+import bio.terra.common.exception.UnauthorizedException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.HttpStatusCodes;
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.model.ErrorReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+/**
+ * SamRetry encapsulates logic needed for retrying Sam API calls. All constants are hard-coded, as
+ * Sam failures are generally just short periods of down-time.
```

Comment:
Also worth noting that this parses/wraps the exceptions and why we think that's a good thing to do.